### PR TITLE
WIP: zxfer: init at 1.1.7

### PR DESCRIPTION
**Pull request by pbsds**
###### Description of changes

Adds the package zxfer.

**Project description**
A popular script in BSD-land for managing ZFS snapshot replication.
It allows the easy and reliable backup, restore or transfer of ZFS filesystems, either locally or remotely.

**Metadata**

* homepage URL: https://github.com/allanjude/zxfer
* source URL: https://github.com/allanjude/zxfer
* license: bsd-2clause
* platforms: unix

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Not yet tested, as I do not have a linux box with ZFS, hence the draft status.

**Question:** should this depend on ZFS?

### CODE REVIEW

File: pkgs/tools/filesystems/zxfer/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{ lib, stdenv, fetchFromGitHub
+, coreutils, gnused, gawk, gnugrep
+, openssh, rsync
+, gzip }:
```
Comment (fabaff) [inline pkgs/tools/filesystems/zxfer/default.nix:4]: One item per line is nowadays the preferred way to do as it creates simpler diffs.

```suggestion
{ lib
, stdenv
, fetchFromGitHub
, coreutils
, gnused
, gawk
, gnugrep
, openssh
, rsync
, gzip
}:
```

File: pkgs/tools/filesystems/zxfer/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "zxfer";
+  version = "1.1.7";
+
+  src = fetchFromGitHub {
+    sha256 = "11SQJcD3GqPYBIgaycyKkc62/diVKPuuj2Or97j+NZY=";
+    rev = "v${version}";
+    repo = "zxfer";
+    owner = "allanjude";
```
Comment (fabaff) [inline pkgs/tools/filesystems/zxfer/default.nix:14]: Sticking to the most common ordering, makes spotting issues easier.

```suggestion
    repo = "zxfer";
    owner = "allanjude";
    rev = "v${version}";
    sha256 = "11SQJcD3GqPYBIgaycyKkc62/diVKPuuj2Or97j+NZY=";
```

File: pkgs/tools/filesystems/zxfer/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  };
+
+  buildInputs = [ coreutils gawk gnused gnugrep openssh rsync ];
+  nativebuildInputs = [ gzip ];
```
Comment (fabaff) [inline pkgs/tools/filesystems/zxfer/default.nix:18]: Also, one per line.

```suggestion
  nativebuildInputs = [
    gzip
  ];
  
  buildInputs = [
    coreutils
    gawk
    gnused
    gnugrep
    openssh
    rsync
  ];
```

File: pkgs/tools/filesystems/zxfer/default.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp zxfer $out/bin/
+
+    mkdir -p $out/share/man/man8/
+    cp zxfer.8 $out/share/man/man8/
```
Comment (fabaff) [inline pkgs/tools/filesystems/zxfer/default.nix:25]: `installManPage` can help here.

File: pkgs/tools/filesystems/zxfer/default.nix [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  '';
+
+  meta = with lib; {
+    inherit version;
```
Comment (fabaff) [inline pkgs/tools/filesystems/zxfer/default.nix:29]: Can be removed.

```suggestion
```

File: pkgs/tools/filesystems/zxfer/default.nix [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    inherit (src.meta) homepage;
```
Comment (fabaff) [inline pkgs/tools/filesystems/zxfer/default.nix:30]: Adding the URL allows one to visit the homepage from the editor.

```suggestion
    homepage = "https://github.com/allanjude/zxfer";
```

File: pkgs/tools/filesystems/zxfer/default.nix [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    description =
+      "Transfer ZFS filesystems, snapshots, properties, files and directories.";
```
Comment (fabaff) [inline pkgs/tools/filesystems/zxfer/default.nix:32]:
```suggestion
    description = "Transfer ZFS filesystems, snapshots, properties, files and directories";
```

see https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes

**Comment by pbsds** [comment]
Closed in favor of https://github.com/NixOS/nixpkgs/pull/208338

**Pull request closed without merging**